ledger/ledger-consensual/src/main/kotlin/net/corda/v5/ledger/consensual/transaction/ConsensualTransactionBuilder.kt

```diff
@@ -0,0 +1,29 @@
+package net.corda.v5.ledger.consensual.transaction
+
+import net.corda.v5.application.crypto.SigningService
+import net.corda.v5.application.serialization.SerializationService
+import net.corda.v5.base.annotations.DoNotImplement
+import net.corda.v5.crypto.DigestService
+import net.corda.v5.crypto.merkle.MerkleTreeFactory
+import net.corda.v5.ledger.common.transactions.SignedTransaction
+import net.corda.v5.ledger.consensual.ConsensualState
+import java.security.PublicKey
+import java.security.SecureRandom
+import java.time.Instant
+
+@DoNotImplement
+interface ConsensualTransactionBuilder {
+    val merkleTreeFactory: MerkleTreeFactory
+    val digestService: DigestService
+    val secureRandom: SecureRandom
+    val serializer: SerializationService
+    val signingService: SigningService
+
+    val timeStamp: Instant?
+    fun withTimeStamp(timeStamp: Instant): ConsensualTransactionBuilder
+
+    val consensualStates: List<ConsensualState>
+    fun withConsensualState(consensualState: ConsensualState) : ConsensualTransactionBuilder
```

Comment:
What do I get if I call `withTimeStamp` or `withConsensualState` -- is this a builder method thing and they return a new `ConsensualTransactionBuilder` copy with those properties added?

Reply:
Yes. https://github.com/corda/corda-runtime-os/pull/1715/files#diff-056e4bffcd08471f085b7fb5255f417ac87f930b4b8bd0552db5e638fa9c2513R61
(Although it is half refactored already to have the services as constructor arguments and return a signed tx, but the concept is roughly the same.)